commands/board/list.go

```diff
@@ -72,39 +71,41 @@ func apiByVidPid(vid, pid string) ([]*rpc.BoardListItem, error) {
 		return nil, errors.Wrap(err, tr("failed to initialize http client"))
 	}
 
-	if res, err := httpClient.Do(req); err == nil {
-		if res.StatusCode >= 400 {
-			if res.StatusCode == 404 {
-				return nil, ErrNotFound
-			}
-			return nil, errors.Errorf(tr("the server responded with status %s"), res.Status)
-		}
-
-		body, _ := ioutil.ReadAll(res.Body)
-		res.Body.Close()
-
-		var dat map[string]interface{}
-		err = json.Unmarshal(body, &dat)
-		if err != nil {
-			return nil, errors.Wrap(err, tr("error processing response from server"))
+	res, err := httpClient.Do(req)
+	if err != nil {
+		return nil, errors.Wrap(err, tr("error querying Arduino Cloud Api"))
+	}
+	if res.StatusCode >= 400 {
```

Comment:
Is this check really required?

Reply:
Yes I guess? the `httpclient.Do` returns an error so why not check it?

BTW the error check was there also before I just moved from the bottom of a very big `if {...}` after inverting the condition from `err == nil` to `err != nil`.  Removing the whitespaces from the diff may help to see it.